core/src/components/button/button.ionic.scss

```diff
@@ -246,26 +267,38 @@
   font-size: globals.$ion-font-size-500;
 }
 
-:host(.button-small) ::slotted(ion-icon[slot="start"]) {
+:host(.button-small) ::slotted(ion-icon[slot="start"]),
+:host(.button-small) ::slotted(ion-spinner[slot="start"]) {
   @include globals.margin-horizontal(null, globals.$ion-space-200);
 }
 
-::slotted(ion-icon[slot="start"]) {
+::slotted(ion-icon[slot="start"]),
+::slotted(ion-spinner[slot="start"]) {
   @include globals.margin-horizontal(null, globals.$ion-space-250);
 }
 
 :host(.button-large) ::slotted(ion-icon[slot="start"]) {
   @include globals.margin-horizontal(null, globals.$ion-space-300);
 }
 
-:host(.button-small) ::slotted(ion-icon[slot="end"]) {
+:host(.button-xlarge) ::slotted(ion-spinner[slot="start"]) {
+  @include globals.margin-horizontal(null, globals.$ion-space-300);
+}
```

Comment:
We can remove this since it can be combined with large as shown in the previous suggestion.

```suggestion
```